pkg/route/validation/validation.go

```diff
@@ -70,8 +73,8 @@ var (
 	permittedResponseHeaderValueRE = regexp.MustCompile(strings.Replace(permittedHeaderValueTemplate, "XYZ", "res", 1))
 )
 
-func ValidateRoute(route *routev1.Route) field.ErrorList {
-	return validateRoute(route, true)
+func ValidateRoute(ctx context.Context, route *routev1.Route, sarCreator routecommon.SubjectAccessReviewCreator, secretsGetter corev1client.SecretsGetter, opts routecommon.RouteValidationOptions) field.ErrorList {
```

Comment:
Passing `context.Context` as the first parameter is go idiomatic, I agree on this. But do we pass a meaningful context from the api server or just `context.Background()`/`context.TODO()`?

Reply:
I think it would be a meaningful context, and not just `context.Background()`/`context.TODO()`. This will be the same context which was previously getting passed in `AllocateHost` and `ValidateHostUpdate`  from api server.

 https://github.com/openshift/openshift-apiserver/blob/master/pkg/route/apiserver/registry/route/strategy.go#L95-L121